src/EditorFeatures/Test/Diagnostics/DiagnosticDataSerializerTests.cs

```diff
@@ -91,10 +91,13 @@ public async Task SerializationTest_Document()
             var version = VersionStamp.Create(utcTime.AddDays(1));
 
             var key = "document";
+
+            var persistentService = workspace.Services.GetRequiredService<IPersistentStorageService>();
             var serializer = new CodeAnalysis.Workspaces.Diagnostics.DiagnosticDataSerializer(analyzerVersion, version);
 
-            Assert.True(await serializer.SerializeAsync(document, key, diagnostics, CancellationToken.None).ConfigureAwait(false));
-            var recovered = await serializer.DeserializeAsync(document, key, CancellationToken.None);
+            Assert.True(await serializer.SerializeAsync(persistentService, document.Project, document, key, diagnostics, CancellationToken.None).ConfigureAwait(false));
```

Comment:
❔ Why pass both `document.Project` and `document`?

Reply:
The document is optional (may be null). The method needs a project even if when document is null.

Reply:
💡 Consider adding a pass-through overload taking a non-null `Document`?

Reply:
Introducing overloads would just add more clutter in this case. The method is called from another one that takes project and optional document. This pattern is quite common in diagnostic code. I think this pattern is fine, but if we want to do something better we should define a struct `DocumentOrProject` that's implicitly convertible from `Document` and `Project` and represents this pair. That would avoid the extra overloads for each method that uses this pattern.